### ISSUE TITLE
GeneralizedPareto.cdf does not consider support

### ISSUE THREAD

**Issue by moudheus** [awslabs/gluon-ts#2345]
## Description
When calling GeneralizedPareto.cdf I expect to get 0 outside of the distribution support and not nan. 

## To Reproduce
```python
import torch
from gluonts.torch.distributions import GeneralizedPareto

xi, beta, z = 2., 1., -1
xit, betat, zt = torch.tensor([[xi]]), torch.tensor([[beta]]), torch.tensor([z])

gp = GeneralizedPareto(xit, betat)

print(gp.log_prob(zt))
print(torch.exp(gp.log_prob(zt)))
print(gp.cdf(zt))
```

## Error message or code output

```
tensor([-inf], dtype=torch.float64)  # this is expected
tensor([0.], dtype=torch.float64)  # this is expected
tensor([nan]) # this is not expected
```

## Environment
- Operating system: ubuntu
- Python version: 3.8.12
- GluonTS version: dev latest
- Torch version: 1.10.0

**Comment by lostella** [comment]
All of those could also raise an exception because the point is not in the support (see e.g. `torch.distributions.Pareto(1, 1)` on negative data)

**Comment by lostella** [comment]
Interestingly, with `validate_args=False` torch's `Pareto` returns `nan` on the above:

```python
pareto = Pareto(torch.tensor([1.0]), torch.tensor([1.0]), validate_args=False)
pareto.log_prob(zt)  # tensor([nan])
pareto.cdf(zt)  # tensor([nan])
```

**Comment by lostella** [comment]
@moudheus I think the best course of action would be to comply with torch behavior, and have all the necessary validation by default (raising an exception in all calls in the above example) and otherwise a not better specified return value if validation is disabled.

Would you agree?

**Comment by moudheus** [comment]
Thank you for the quick response. I agree for the distribution parameters, they have a validity domain and it makes sense to throw an exception outside the domain. For the input of the pdf or cdf it is different, generally the validity domain is -inf to +inf. The support is where the pdf is 0, not the validity domain. So outside the support I would prefer the log_prob to output -inf and the cdf to output the proper value. I understand that pytorch and tensorflow throw exceptions outside the support (when validate=True), I guess it is because of the predominant usage of log likelihood that does not like probabilities being 0, but other loss functions do not have this problem and would benefit from not having these "not in support" exceptions thrown. It can be fixed by the caller but the caller has to know the support which can depend on the parameters. Anyway this is not a blocking pb for me, I fixed this by calling torch.clamp(x, 0.) in the cdf.
